airbyte-cdk/python/airbyte_cdk/sources/streams/http/error_handlers/response_models.py

```diff
@@ -22,10 +26,27 @@ class ErrorResolution:
     error_message: Optional[str] = None
 
 
-DEFAULT_ERROR_RESOLUTION = ErrorResolution(
-    response_action=ResponseAction.RETRY,
-    failure_type=FailureType.system_error,
-    error_message="The request failed due to an unknown error.",
-)
+def _format_exception_error_message(exception: Exception) -> str:
+    return f"{type(exception).__name__}: {str(exception)}"
+
+
+def _format_response_error_message(response: requests.Response) -> str:
+    try:
+        response.raise_for_status()
+    except HTTPError as exception:
+        return filter_secrets(f"Response was not ok: `{str(exception)}`. Response content is: {response.text}")
+    # We purposefully do not add the response.content because the response is "ok" so there might be sensitive information in the payload.
+    # Feel free the
+    return f"Unexpected response with HTTP status {response.status_code}"
+
+
+def create_fallback_error_resolution(response_or_exception: Union[requests.Response, Exception]) -> ErrorResolution:
```

Comment:
Note that the interface does not accept `None` like the `interpret_response` does. I started [a conversation in our private slack](https://airbytehq-team.slack.com/archives/C02U9R3AF37/p1723128412435789). Basically:

In which case can [response_or_exception](https://github.com/airbytehq/airbyte/blob/99a33cb90319057ea75264b5a7b5247ae56189a5/airbyte-cdk/python/airbyte_cdk/sources/streams/http/error_handlers/error_handler.py#L33) be None? The typing says that it is optional but [the only call where response_or_exception is not a passthrough](https://github.com/airbytehq/airbyte/blob/99a33cb90319057ea75264b5a7b5247ae56189a5/airbyte-cdk/python/airbyte_cdk/sources/streams/http/http_client.py#L257)(https://github.com/airbytehq/airbyte/blob/99a33cb90319057ea75264b5a7b5247ae56189a5/airbyte-cdk/python/airbyte_cdk/sources/streams/http/http_client.py#L257) does not seem to pass `None`.

I’m asking because:
* I’m not sure what it would mean
* It’s another case to handle when dealing with the parameter `response_or_exception`
* Most error handlers do not seem to handle which seems dangerous for `AttributeError: 'NoneType' object has no attribute X`

Reply:
Until we figure this out, this will generate mypy issues as seen [here](https://github.com/airbytehq/airbyte/actions/runs/10307336097/job/28532403285?pr=43399)

Reply:
Based on the Slack thread, is the proposal to modify the `interpret_response()` method interface to no longer be `Optional[blahblah]`? And that would be the solve for typing isses since `interpret_response()` -> `create_fallback_error_resolution()` will be passing the union now.

I'm not opposed to that, but are worried about all the backwards compatibility headaches? since we do have some uses of the various `DefaultErrorHandlers` that technically still adhere to the same interface?

Would the alternative to the mypy fix be to add a type check or None check around the calls to `create_fallback_error_resolution()` to guarantee it's not `None`, even though we're confident it never is. I didn't test myself, but I'm curious if that would make mypy stop complaining

Or the alternative alternative is just an `ignore` lol

Reply:
Agreed that it is a bigger change. Let's decouple the typing change from this change then